l10n_it_invoices_data_communication/models/communication.py

```diff
@@ -284,14 +288,13 @@ def _prepare_cedente_partner_id(self, partner, vals=None):
         # ----- Get fiscalcode
         vals['cedente_CodiceFiscale'] = \
             partner.commercial_partner_id.fiscalcode or ''
-        vals['cedente_Denominazione'] = partner.name.encode('utf8') or ''
+        vals['cedente_Denominazione'] = self.encode(partner.name, 80)
         # Sede
         vals['cedente_sede_Indirizzo'] = '{} {}'.format(
-            partner.street and partner.street.encode('utf8') or '',
-            partner.street2 and partner.street2.encode('utf8') or '').strip()
-        vals['cedente_sede_Cap'] = partner.zip or ''
-        vals['cedente_sede_Comune'] = partner.city and \
-            partner.city.encode('utf8') or ''
+            self.encode(partner.street or '', 60),
+            self.encode(partner.street2 or '', 60)).strip()
+        vals['cedente_sede_Cap'] = self.encode(partner.zip or '', 5, encoding='ascii')
```

Comment:
lo zip dovrebbe essere un formato creato appositamente, non un ascii, lo trovi nel dizionario (forse 5 numeri banalmente)

Reply:
Ciao Sergio,
il formato prevede 5 caratteri BasicLatin, ovvero ASCII, infatti lo trovi nel XSD

![image](https://user-images.githubusercontent.com/16492632/67751601-b34cb100-fa31-11e9-8db2-f5752875cb86.png)

Lo so che nel caso in cui il CAP provenga da Odoo non dovrebbero esserci mai problemi, ma nel solito caso dei dati provenienti da copia/incolla da pagine web non possiamo mai essere certi della presenza o meno di sole cifre...

Reply:
c'è questo nel bindings creato con pyxb:
`CAPType._CF_pattern.addPattern(pattern='[0-9][0-9][0-9][0-9][0-9]')`
a me sembra che accetti solo 5 numeri, probabilmente c'è una regola da qualche altra parte nell'xsd?

Reply:
@sergiocorato ti confondi con quello della fattura elettronica? `l10n_it_invoices_data_communication` non usa `pyxb`

Reply:
@eLBati davo per scontato che avessero tenuto lo stesso formato, invece solo per la fattura elettronica è così, quindi ok